src/openms/source/QC/DBSuitability.cpp

```diff
@@ -101,102 +91,51 @@ namespace OpenMS
     fdr.setParameters(p);
     fdr.apply(pep_ids);
 
-    for (PeptideIdentification& pep_id : pep_ids)
-    {
-      // sort hits by q-value
-      pep_id.sort();
-
-      vector<PeptideHit>& hits = pep_id.getHits();
-
-      if (hits.empty()) continue;
-
-      const PeptideHit& top_hit = hits[0];
-
-      // skip if the top hit is a decoy hit
-      if (!top_hit.metaValueExists("target_decoy"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No target/decoy information found! Make sure 'PeptideIndexer' is run beforehand."));
-      }
-      if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
-
-      // skip if top hit is out ouf FDR
-      if (!passesFDR_(top_hit, FDR)) continue;
-
-      // check if top hit is found in de novo protein
-      if (!isNovoHit_(top_hit)) // top hit is db hit
-      {
-        ++data.num_top_db;
-        continue;
-      }
-
-      // find the second target hit, skip all decoy or novo hits inbetween
-      const PeptideHit* second_hit = nullptr;
-      for (UInt i = 1; i < hits.size(); ++i)
-      {
-        // check for FDR
-        if (!passesFDR_(hits[i], FDR)) break;
-
-        // check if target, also check for "target+decoy" value
-        String td_info(hits[i].getMetaValue("target_decoy"));
-        if (td_info.find("target") != 0) continue;
-
-        // check if hit is novo hit
-        if (isNovoHit_(hits[i])) continue;
-
-        second_hit = &hits[i];
-        break;
-      }
-      if (second_hit == nullptr) // no second target hit with given FDR found
-      {
-        ++data.num_top_novo;
-        continue;
-      }
-
-      // second hit is db hit
-      ++data.num_interest;
-
-      // check for re-ranking
-      if (no_rerank)
-      {
-        ++data.num_top_novo;
-        continue;
-      }
-
-      // check for xcorr score
-      if (!top_hit.metaValueExists("MS:1002252") || !second_hit->metaValueExists("MS:1002252"))
-      {
-        throw(Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "No cross correlation score found at peptide hit. Only Comet search engine is supported right now."));
-      }
-
-      double top_xscore_mw = double(top_hit.getMetaValue("MS:1002252")) / top_hit.getSequence().getMonoWeight();
-      double second_xscore_mw = double(second_hit->getMetaValue("MS:1002252")) / second_hit->getSequence().getMonoWeight();
-      if (top_xscore_mw - second_xscore_mw <= data.cut_off)
-      {
-        ++data.num_top_db;
-        ++data.num_re_ranked;
-      }
-      else
-      {
-        ++data.num_top_novo;
-      }
-    }
-
-    if (data.num_top_db == 0 && data.num_top_novo == 0)
-    {
-      OPENMS_LOG_WARN << "Identifications could not be assigned to either the database or the deNovo protein. Probably your FDR threshold is too strict." << endl;
-      data.suitability = DBL_MAX;
-      return;
-    }
+    // calculate suitability
+    SuitabilityData d;
+    results_.push_back(d);
```

Comment:
`results_.push_back(SuitabilityData());` avoids accidentally referring back to `d`, which would be a mistake